boards/hail/src/main.rs

```diff
@@ -53,12 +54,13 @@ struct Hail {
     console: &'static Console<'static, usart::USART>,
     gpio: &'static capsules::gpio::GPIO<'static, sam4l::gpio::GPIOPin>,
     timer: &'static TimerDriver<'static, VirtualMuxAlarm<'static, sam4l::ast::Ast<'static>>>,
-    isl29035: &'static capsules::isl29035::Isl29035<'static,
-                                                    VirtualMuxAlarm<'static,
-                                                                    sam4l::ast::Ast<'static>>>,
-    si7021: &'static capsules::si7021::SI7021<'static,
-                                              VirtualMuxAlarm<'static, sam4l::ast::Ast<'static>>>,
-    ninedof: &'static capsules::ninedof::NineDof<'static>,
+    //isl29035: &'static capsules::isl29035::Isl29035<'static,
+    //                                                VirtualMuxAlarm<'static,
+    //                                                                sam4l::ast::Ast<'static>>>,
+    //si7021: &'static capsules::si7021::SI7021<'static,
+    //                                          VirtualMuxAlarm<'static, sam4l::ast::Ast<'static>>>,
+    //ninedof: &'static capsules::ninedof::NineDof<'static>,
+    //i2c_master_slave: &'static capsules::i2c_master_slave_driver::I2CMasterSlaveDriver<'static>,
```

Comment:
This is a problem we've started running into and are not going to escape anytime soon. There are some pins in a platform that absolutely do not make sense to hard-code configure at compile-time rather than configure based on which applications are loaded at boot.

Ideally, we would have Signbus, at initialization time, attempt to take a lock on four GPIO pins, configuring two as GPIOs and two as I2C. If it could not take a lock (because a different application was already using them for instance) it would fail and report that problem. I'm unsure what the best way to implement this kind of thing would be though, especially in a way that is chip agnostic. @alevy Do you have any thoughts as to how best to do this?

D4 and D5 could be changed to be I2C. They are currently configured as SPI (along with D2/D3), but this was an arbitrary choice, and we could make a different arbitrary choice for now if it makes things easier.

Alternatively, we could make this a new platform (something like Hail-Signbus maybe), but that starts wandering down a path of dozens of nearly identical platforms, which feels dangerous and bad.